### PR TITLE
Switched to FF50 and added fixes for stale element exceptions

### PR DESCRIPTION
**Pull request by viorelaioia**

### CODE REVIEW

**Comment by viorelaioia** [comment]
Started adhoc run: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/mozillians.adhoc/56/console

**Comment by viorelaioia** [comment]
latest adhoc run: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/mozillians.adhoc/96/

**Comment by viorelaioia** [comment]
@davehunt , can you please take a look over the changes in this PR? Thanks!

File: tests/test_profile.py [hunk 1]
```diff
@@ -86,11 +86,10 @@ def test_skill_deletion(self, base_url, selenium, vouched_user):
         skills_form.click_update()
 
         settings = home_page.header.click_settings_menu_item()
+        # TODO: Try to remove the sleep after https://bugzilla.mozilla.org/show_bug.cgi?id=1291320 is fixed.
```
Comment (stephendonner) [inline tests/test_profile.py]: For this, might I recommend also filing an Issue in this GitHub repo noting to return to this?  Always good to have it visible (no pun intended) :-)

File: pages/edit_group.py [hunk 2]
```diff
@@ -16,6 +17,12 @@ class EditGroupPage(Base):
     _invitations_button_locator = (By.ID, 'invitations-tab')
     _invitations_tab_locator = (By.ID, 'invitations')
 
+    def __init__(self, base_url, selenium):
+        Base.__init__(self, base_url, selenium)
```
Comment (johngian) [inline pages/edit_group.py]: Please replace this with:

``super(EditGroupPage, self).__init__(base_url, selenium)``

File: pages/register.py [hunk 3]
```diff
@@ -71,7 +71,10 @@ def check_recaptcha(self):
         )
         self.selenium.switch_to_default_content()
 
-    def click_create_profile_button(self):
+    def click_create_profile_button(self, leavepage=True):
         self.selenium.find_element(*self._create_profile_button_locator).click()
-        from pages.profile import Profile
-        return Profile(self.base_url, self.selenium)
+        if leavepage is False:
```
Comment (johngian) [inline pages/register.py]: Please replace this with ``if not leavepage``.

File: pages/edit_group.py (hunk 2 above)
Comment (johngian) [inline pages/edit_group.py]: This might have other occurrences across the codebase.

**Comment by johngian** [comment]
Please rebase against master.

**Comment by viorelaioia** [comment]
http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/mozillians.adhoc/107/console

File: pages/edit_group.py (hunk 2 above)
Comment (viorelaioia) [inline pages/edit_group.py]: Updated.

File: tests/test_profile.py (hunk 1 above)
Comment (viorelaioia) [inline tests/test_profile.py]: @stephendonner , I opened https://github.com/mozilla/mozillians-tests/issues/283

File: pages/register.py (hunk 3 above)
Comment (viorelaioia) [inline pages/register.py]: done

**Comment by johngian** [comment]
Please rebase so we only have 1 commit.

**Pull request merged**